gdb/kllvm.py

```diff
@@ -630,19 +640,144 @@ def append(self, subject, isVar, sort):
+            return None
+        kind = s.name
+        sort = val.type.name + '{}'
+
+    if kind == "block" or kind == "string":
+        return termPrinter(val, "block", sort)
+    elif kind == "list":
+        return termPrinter(val, "list", sort)
+    elif kind == "map":
+        return termPrinter(val, "map", sort)
+    elif kind == "set":
+        return termPrinter(val, "set", sort)
+    elif kind == "stringbuffer":
+        return termPrinter(val, "stringbuffer", sort)
+    elif kind == "__mpz_struct":
+        return termPrinter(val, "int", sort)
+    elif kind == "floating":
+        return termPrinter(val, "floating", sort)
     return None
+
+class KPrefix(gdb.Command):
+    "Generic command for stepping through a K framework semantics."
+
+    def __init__(self):
+        super(KPrefix, self).__init__("k", gdb.COMMAND_NONE, gdb.COMPLETE_COMMAND, True)
+
+prefix = KPrefix()
+
+class KStart(gdb.Command):
+    "Start the program but do not take any steps."
+
+    def __init__(self):
+        super(KStart, self).__init__("k start", gdb.COMMAND_RUNNING, gdb.COMPLETE_NONE)
+
+    def invoke(self, arg, from_tty):
+        gdb.execute("start", from_tty)
+        gdb.execute("advance definition.kore:step", from_tty)
+
+start = KStart()
+
+class KStep(gdb.Command):
+    "Take a specified number of rewrite steps."
+
+    def __init__(self, name, cat):
+        super(KStep, self).__init__(name, cat, gdb.COMPLETE_NONE)
+
+    def invoke(self, arg, from_tty):
+        times = 1
+        if arg != "":
+            times = int(arg)
+        if times > 0:
+            bp = gdb.Breakpoint("definition.kore:step", internal=True, temporary=True)
+            bp.ignore_count = times-1
+            gdb.execute("c", from_tty)
+        else:
+            gdb.execute("step 0", from_tty)
+
+step = KStep("k step", gdb.COMMAND_RUNNING)
+step2 = KStep("k s", gdb.COMMAND_NONE)
+
+class KMatch(gdb.Command):
+    """Attempt to match a particular rule against a particular configuration.
+
+Takes two arguments: The first is a top-level rule label.
+The second is a term of sort GeneratedTopCell. For example, from the "step"
+function, you could say something like "k match MODULE.LABEL subject"
+to try to match the specified rule against the current configuration.
+
+Does not actually take a step if matching succeeds.
+    """
+
+    def __init__(self):
+        super(KMatch, self).__init__("k match", gdb.COMMAND_RUNNING, gdb.COMPLETE_SYMBOL)
+
+    def invoke(self, arg, from_tty):
+        try:
+            argv = gdb.string_to_argv(arg)
+            gdb.lookup_global_symbol("resetMatchReason").value()()
+            gdb.lookup_global_symbol(argv[0] + '.match').value()(gdb.parse_and_eval(argv[1]))
+            entries = gdb.lookup_global_symbol("getMatchLog").value()()
+            size = int(gdb.lookup_global_symbol("getMatchLogSize").value()())
+            for i in range(size):
+                entry = entries[i]
+                if entry['kind'] == 0:
```

Comment:
The c++ code uses an enum for these values, some python constants with the same name would be nice.